docroot/themes/humsci/humsci_basic/src/js/main-nav/nested-toggler.js

```diff
@@ -0,0 +1,72 @@
+const togglers = document.querySelectorAll('.hb-nested-toggler');
+const mobileNavBreakpoint = 992;
+let windowWidth = window.innerWidth;
+
+if (togglers) {
+  for (let i = 0; i < togglers.length; i += 1) {
+    const toggler = togglers[i];
+    const togglerID = toggler.getAttribute('id');
+    const togglerContent = document.querySelector(`[aria-labelledby="${togglerID}"]`);
+    const togglerParent = toggler.closest('.su-main-nav__item--parent');
+    const shouldBeOpen = togglerParent.classList.contains('su-main-nav__item--active-trail') ? true : false;
+    let isHidden;
+
+    const collapseMenu = () => {
+      togglerContent.setAttribute('aria-hidden', true);
+      toggler.setAttribute('aria-expanded', false);
+      isHidden = true;
+    }
+
+    const showMenu = () => {
+      togglerContent.setAttribute('aria-hidden', false);
+      toggler.setAttribute('aria-expanded', true);
+      isHidden = false;
+    }
+
+    // On page load:
+    // - All menus in the active section should be open
+    // - All other menus should be hidden
+    if (shouldBeOpen && windowWidth < mobileNavBreakpoint) {
+      showMenu();
+    } else {
+      collapseMenu();
+    }
+
+    toggler.addEventListener('click', (e) => {
+      e.preventDefault();
+
+      // Toggle the aria-hidden and aria-expanded values on click
+      if (isHidden) {
+        showMenu();
+      } else {
+        collapseMenu();
+      }
+    });
+
+    // At larger screen sizes:
+    // =========================================================================
+    // All menus collapse when resizing larger than the lg breakpoint
+    window.addEventListener('resize', () => {
+      windowWidth = window.innerWidth;
+
+      // When resizing from mobile to desktop, show the navigation
+      if (windowWidth >= mobileNavBreakpoint) {
```

Comment:
youll need a wasDesktopSize here too so collapseMenu doesn't keep getting called. I think the logic would go like this:


```
if (windowWidth >= mobileNavBreakpoint && !wasDesktopSize) {
  collapseMenu();
  wasDesktopSize = true;
}

```

but I wonder if you need this since we'll close the nav on desktop in the menu-toggle.js?

Reply:
Adding this creates a new edge case when resizing the browser. Personally I think this edge case happening at desktops isn't a huge issue but the new issue it creates is. I would suggest we skip this and if for some reason the edge case is a problem in the future we can tackle it then.

Reply:
that is fine with me!! not a huge deal :)